packages/vats/src/bootstrap.js

```diff
@@ -945,32 +945,40 @@ export function buildRootObject(vatPowers, vatParameters) {
 
         // ag-setup-solo runs this.
         case 'client': {
-          assert(FIXME_GCI, X`client must be given GCI`);
+          let localBundle;
+          let chainBundle;
+          const deprecated = {};
 
-          const localTimerService = await E(vats.timer).createTimerService(
-            devices.timer,
-          );
-          await registerNetworkProtocols(vats, bridgeManager, null);
+          // Tell the http server about our presences.
+          const updatePresences = () =>
+            E(vats.http).setPresences(localBundle, chainBundle, deprecated);
 
-          await setupCommandDevice(vats.http, devices.command, {
-            client: true,
-          });
-          await addRemote(FIXME_GCI);
-          // addEgress(..., index, ...) is called in vat-provisioning.
-          const demoProvider = await E(vats.comms).addIngress(
-            FIXME_GCI,
-            PROVISIONER_INDEX,
-          );
-          const localBundle = await createLocalBundle(
-            vats,
-            devices,
-            vatAdminSvc,
-          );
-          await E(vats.http).setPresences(localBundle);
-          const bundle = await E(demoProvider).getDemoBundle();
-          await E(vats.http).setPresences(localBundle, bundle, {
-            localTimerService,
-          });
+          const addLocalPresences = async () => {
+            await registerNetworkProtocols(vats, bridgeManager, null);
+
+            await setupCommandDevice(vats.http, devices.command, {
+              client: true,
+            });
+            localBundle = await createLocalBundle(vats, devices, vatAdminSvc);
+
+            // TODO: Remove this when we can.
+            deprecated.uploads = localBundle.scratch;
+            await updatePresences();
```

Comment:
I'm a bit weirded out that we've got two calls to `updatePresences()` that are racing against each other. I guess it's not much weirder than the original that called `updatePresences()` twice in quick succession, with just the `getDemoBundle()` call in between. But it'd be nice if there were an easy way to remove the duplicate: I had to read `vat-http.js` to convince myself that multiple calls (in an unspecified order) wouldn't cause any problems.

Reply:
> it'd be nice if there were an easy way to remove the duplicate

The duplicate is kinda the whole raison d'être of this PR.  That way, the local presences on `home` appear and can be used, even before the chain has loaded.  I saw no reason to force that sequence, though, since decoupling them has nicer properties in event of a failure.

I will write a more explicit comment describing what happens, though, to help the poor reader (such as future me) see what is going on.